src/components/contexts/UserContext.tsx

```diff
@@ -62,46 +59,89 @@ export function UserContextProvider({ children }: Props) {
           signOut();
           console.error("User not in group");
         }
+        if (!user.userSub) {
+          throw new Error("No user");
+        }
 
-        const response = await client.models.User.get({
-          id: user.tokens?.accessToken.payload.username as string,
-        });
+        try {
+          const response = await client.models.User.get({
+            id: user.userSub as string,
+          });
 
-        if (response.data === null) {
-          // Logout User record does not exist in DB
-          signOut();
-          console.error("User not in DB");
-        }
+          if (response.data === null) {
+            // Logout User record does not exist in DB
+            signOut();
+            console.error("User not in DB");
+          }
 
-        setCurrentUser({
-          username: user.tokens?.accessToken.payload.username as string,
-          type: (
-            user.tokens?.idToken?.payload["cognito:groups"] as UserType[]
-          )?.[0],
-          populated: true,
-          completedProfile: response.data?.completedRegistration ?? false,
-          email: response.data?.email ?? "",
-          firstName: response.data?.firstName ?? "",
-          lastName: response.data?.lastName ?? "",
-        });
-      } catch (err) {
-        if (String(err).includes("No user")) {
-          setCurrentUser({
+          return {
+            username: user.tokens?.accessToken.payload.username as string,
+            type: (
+              user.tokens?.idToken?.payload["cognito:groups"] as UserType[]
+            )?.[0],
+            populated: true,
+            completedProfile: response.data?.completedRegistration ?? false,
+            email: response.data?.email ?? "",
+            firstName: response.data?.firstName ?? "",
+            lastName: response.data?.lastName ?? "",
+          } as IUser;
+        } catch (error) {
+          console.error(error);
+        }
+        // Set user information based on the authentication session...
+      } catch (error) {
+        if (String(error).includes("No user")) {
+          console.info("Not Logged in");
+          return {
             username: "",
             type: UserType.Guest,
             populated: true,
-          });
-          console.info("Not Logged in");
+          } as IUser;
         } else {
-          console.error(err);
+          console.error(error);
         }
       }
-    }
-    void currentAuthenticatedUser();
+    },
+  });
+
+  useEffect(() => {
```

Comment:
Hub initialized at render to listen to sign in sign out auth events and there are more use cases here just in case:  https://docs.amplify.aws/react/build-a-backend/auth/connect-your-frontend/listen-to-auth-events/

In case we need to listen to token refreshses, login redirects, etc